gdrive

```diff
@@ -235,23 +250,28 @@ def main():
 
     service = init()
     if service is None:
-        print('Failed to start service. Exiting..')
+        print('Failed to start service. Exiting...')
         return
 
     opt = read_argv()
-    # If there is an ID, downloads with that ID
-    # If there is a name, downloads with that name
-    # If there is nothing, downloads with the third argument
     if func == 'download':
+        # Downloads with the given file ID
         if opt['file_id']:
             download(service, opt['file_id'])
+        # --last overrides --name if true
+        elif opt['file_name'] or opt['download_last']:
+            download_name(service, opt['file_name'], opt['download_last'])
```

Comment:
This is too confusing. You're using the "download_last" as a "list_all" which aren't the same thing conceptually.

Reply:
Create another function to deal with each option.

Reply:
The code is confusing, but I highly disagree, as the only change is a boolean passed as true.
Although to try to improve that, I made some changes to how it is handled and renamed 'download_name' to 'download_search' and 'list_all' to 'download_last'.

Reply:
I put a comment below regarding this discussion.